src/geometry/CTiglTransformation.h

```diff
@@ -125,6 +128,17 @@ class CTiglTransformation
     // Getter for matrix values
     TIGL_EXPORT double GetValue(int row, int col) const;
 
+    // Set rotation and scaling to trivial values,
+    // the affine part will be the identity,
+    // the translation part will remain unchanged.
+    TIGL_EXPORT void SetTrivialScalingAndRotation();
+
+    // Return the transformation that bring the vector A in the same direction as vector B
+    TIGL_EXPORT CTiglTransformation static GetRotationToAlignAToB(tigl::CTiglPoint vectorA, tigl::CTiglPoint vectorB);
```

Comment:
I don't think that this function should belong here!

This is a very specific function for a very general class.

Also, static and CTiglTransformation has to be swapped to be valid c++.

Reply:
Ok, I can implement the function "GetRotationToAlignAToB" in "tiglmathfunction" and then use this rotation to create a CTiglTransformation. Would this solution be better from your point of view?

Reply:
> Also, static and CTiglTransformation has to be swapped to be valid c++.
This is actually not true. I had to discover this very recently, that type ids, qualifiers and specifiers can appear in any order. See https://en.cppreference.com/w/cpp/language/declarations under Specifier:
> Declaration specifiers (decl-specifier-seq) is a sequence of the following whitespace-separated specifiers, in any order: ...

Thus the following declaration may look like madness, but is perfectly valid:
`const static volatile CTiglTransformation constexpr inline f(int arg);`

Reply:
Okay well. At least swig says, that it is not valid :smile: 

From a practical point of view, it has to be changed.

Reply:
> Ok, I can implement the function "GetRotationToAlignAToB" in "tiglmathfunction" and then use this rotation to create a CTiglTransformation. Would this solution be better from your point of view?

Please implement it in tiglcommonfunctions.h/cpp.

Reply:
In fact, I am not totally convinced.
Because, to make the rotation, the most easy way that I found , is to use matrices. So, the function to create the rotation will probably use matrices, then it make sense to create directly a function that create a CTiglTransformation and in this case, for me, it more usefully to but this in CTiglTransformation class, because a new developer will look for CTiglTransformation functions in CTiglTransformation class and not in tiglcomonfunction. But if you want I can put in tiglcommonfunction.